doc/TROUBLESHOOTING.md

```diff
@@ -145,3 +145,69 @@ Mountpoint logs should contain an error similar to below:
 WARN setattr{req=4 ino=21 name="init.txt"}:
 mountpoint_s3::fuse: setattr failed: inode error: inode 21 (full key "init.txt") is a remote inode and its attributes cannot be modified
 ```
+
+## Invalid Hostname for DNS resolution
+
+Mountpoint by default resolves endpoint for requests in [virtual hosted style](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html).
+If your storage provider does not support virtual style hosted bucket, you may recieve the following error:
+
+```
+Error: Failed to create S3 client
+
+Caused by:
+    0: initial ListObjectsV2 failed for bucket my-bucket in region us-east-1
+    1: Client error
+    2: Unknown CRT error
+    3: CRT error 1059: aws-c-io: AWS_IO_DNS_INVALID_NAME, Host name was invalid for dns resolution.
+Error: Failed to create mount process
+```
+
+In this case, try using `--force-path-style` CLI option when you are mounting the bucket using Mountpoint.
+
+NOTE - Third party storage provider are not officially supported by Mountpoint for Amazon S3.
+
+## Directory disappear after deleting all the files within it
+
+Amazon S3 does not support directories and objects are just grouped using prefix.
+Mountpoint automatically infers a directory structure for your bucket by treating the `/` separator after prefix in your object keys as a delimiter between directories.
+
+So, if all the files within a prefix is deleted, the prefix itself cease to exist.
+In this case, it is expected that mountpoint will not be able to show the directory for listing or other file system operation.
+
+Workaround to persist a directory could be creating a hidden empty file (for example, `.keep`).
```

Comment:
I'd like documentation to be clear here that MP does not hide or treat in any special way those `.keep` files (nor does S3). It's `ls` who filter them out without an `-a` option, right?